onnxruntime/core/providers/cpu/tensor/scatter_nd.cc

```diff
@@ -263,6 +272,84 @@ struct Func_Mul_ND<BFloat16> {
   }
 };
 
+template <class T>
+struct Func_Min_ND {
+  void operator()(T* a, const T* b, uint64_t element_to_copy) const {
+    while (element_to_copy-- > 0) {
+      (*a) = (*a) < (*b) ? (*a) : (*b);
+      a++;
+      b++;
+    }
+  }
+};
+
+template <>
+struct Func_Min_ND<bool> {
+  void operator()(bool*, const bool*, uint64_t) const {
+    ORT_NOT_IMPLEMENTED("CPU execution provider: bool data type is not supported with ScatterND opset 18 when reduction is 'min'.");
+  }
+};
+
+template <>
+struct Func_Min_ND<std::string> {
+  void operator()(std::string*, const std::string*, uint64_t) const {
+    ORT_NOT_IMPLEMENTED("CPU execution provider: string data type is not supported with ScatterND opset 18 when reduction is 'min'.");
+  }
+};
+
+template <>
+struct Func_Min_ND<MLFloat16> {
+  void operator()(MLFloat16*, const MLFloat16*, uint64_t) const {
+    ORT_NOT_IMPLEMENTED("CPU execution provider: MLFloat16 data type is not supported with ScatterND opset 18 when reduction is 'min'.");
+  }
+};
+
+template <>
+struct Func_Min_ND<BFloat16> {
+  void operator()(BFloat16*, const BFloat16*, uint64_t) const {
+    ORT_NOT_IMPLEMENTED("CPU execution provider: BFloat16 data type is not supported with ScatterND opset 18 when reduction is 'min'.");
+  }
+};
+
+template <class T>
+struct Func_Max_ND {
+  void operator()(T* a, const T* b, uint64_t element_to_copy) const {
+    while (element_to_copy-- > 0) {
+      (*a) = (*a) > (*b) ? (*a) : (*b);
+      a++;
+      b++;
+    }
+  }
+};
+
+template <>
+struct Func_Max_ND<bool> {
+  void operator()(bool*, const bool*, uint64_t) const {
+    ORT_NOT_IMPLEMENTED("CPU execution provider: bool data type is not supported with ScatterND opset 18 when reduction is 'max'.");
```

Comment:
For my own education. Why are these types unsupported for "Min/Max"?

Reply:
I think Min/Max only apply to numeric data types. It does not make sense, for example, to do Max(true, false).

Reply:
Hmmm. I guess it is a question for the ONNX spec to clarify. If this is the case, the ONNX spec should be updated. Min/Max is ambiguous for bool: we could interpret false/true as 0/1 and give it a meaning.

Reply:
And why is `MLFloat16` and `BFloat16` skipped?

And regarding Min/Max for bool, the convention seems to interpret it as 0/1 as Rama mentioned.
```python
>>> max(True, False)
True
>>> import torch
>>> torch.max(torch.tensor([True, False]))
tensor(True)
>>> import numpy
>>> numpy.max(numpy.array([True, False]))
True
```

Reply:
16 float is ingeneral not supported with cpu provider.
re false/true to 0/1, I think it is better to add OR/And/Xor logical reduction mothod